src/qat/purr/compiler/execution.py

```diff
@@ -468,7 +468,10 @@ def create_duration_timeline(self, instructions: List[QuantumInstruction]):
         # Strip timelines that only hold delays, since that just means nothing is
         # happening on this channel.
         for key, timeline in dict(results).items():
```

Comment:
minor: instead of a new dict:

```
dict = {k, v for k, v in dict.items() if somthing_about(k, v)}
```

Reply:
Done, with the formatting it does take up more lines tho